scvi/external/sysvi/_model.py

```diff
@@ -0,0 +1,336 @@
+        self.init_params_ = self._get_init_params(locals())
+
+        logger.info("The model has been initialized")
+
+    @torch.no_grad()
+    def get_latent_representation(
+        self,
+        adata: AnnData,
+        indices: Sequence[int] | None = None,
+        cycle: bool = False,
+        give_mean: bool = True,
+        batch_size: int | None = None,
+        as_numpy: bool = True,
+    ) -> np.ndarray | torch.Tensor:
+        """Return the latent representation for each cell.
+
+        Parameters
+        ----------
+        adata
+            Input adata for which latent representation should be obtained.
+        indices
+            Data indices to embed. If None embedd all cells.
+        cycle
+            Return latent embedding of the cycle pass.
+        give_mean
+            Return the posterior mean instead of a sample from the posterior.
+        batch_size
+            Minibatch size for data loading into model. Defaults to `scvi.settings.batch_size`.
+        as_numpy
+            Return in numpy rather than torch format.
+
+        Returns
+        -------
+        Latent Embedding
+        """
+        # Check model and adata
+        self._check_if_trained(warn=False)
+        # TODO extend to check if adata setup is correct wrt training data
+        adata = self._validate_anndata(adata)
+        if indices is None:
+            indices = np.arange(adata.n_obs)
+        # Prediction
+        # Do not shuffle to retain order
+        tensors_fwd = self._make_data_loader(
+            adata=adata, indices=indices, batch_size=batch_size, shuffle=False
+        )
+        predicted = []
+        for tensors in tensors_fwd:
+            # Inference
+            inference_inputs = self.module._get_inference_input(tensors)
+            inference_outputs = self.module.inference(**inference_inputs)
+            if cycle:
+                selected_system = self.module.random_select_systems(tensors["system"])
+                generative_inputs = self.module._get_generative_input(
+                    tensors,
+                    inference_outputs,
+                    selected_system=selected_system,
+                )
+                generative_outputs = self.module.generative(
+                    **generative_inputs, x_x=False, x_y=True
+                )
+                inference_cycle_inputs = self.module._get_inference_cycle_input(
+                    tensors=tensors,
+                    generative_outputs=generative_outputs,
+                    selected_system=selected_system,
+                )
+                inference_outputs = self.module.inference(**inference_cycle_inputs)
+            if give_mean:
+                predicted += [inference_outputs["z_m"]]
+            else:
+                predicted += [inference_outputs["z"]]
+
+        predicted = torch.cat(predicted)
+
+        if as_numpy:
+            predicted = predicted.cpu().numpy()
```

Comment:
```suggestion
        predicted = torch.cat(predicted).cpu()

        if as_numpy:
            predicted = predicted.numpy()
```
So that the tensor is on CPU if `as_numpy=False`